XTS/WS-T/dev/src/org/jboss/jbossts/xts/recovery/participant/at/XTSATRecoveryManager.java

```diff
@@ -29,6 +29,10 @@ public static XTSATRecoveryManager getRecoveryManager()
             // theRecoveryManager must eventually be set to a non-null value
             while (theRecoveryManager == null) {
                 try {
+                    
+                    if (Thread.currentThread().isInterrupted()) {
+                        break;
```

Comment:
This change means that a null value could be returned but the method is designed to guarantee that a non-null value is returned so other things will break.

Something must eventually call setRecoveryManager() which will call notifyAll causing this thread to return from the call to `lock.wait();` and loop test will be re-evaluated.

The thread synchronisation is enforced by running the relevant code inside a `synchronized (lock)` code block.

Reply:
Thanks for the review @mmusgrov . I removed that block of code, now the method guarantees that a non-null value is returned